src/darwin/Framework/CHIP/CHIPControllerAccessControl.mm

```diff
@@ -0,0 +1,75 @@
+/**
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#import "CHIPControllerAccessControl.h"
+
+#include <access/AccessControl.h>
+#include <access/Privilege.h>
+#include <access/RequestPath.h>
+#include <access/SubjectDescriptor.h>
+#include <app-common/zap-generated/ids/Clusters.h>
+#include <lib/core/CHIPError.h>
+
+using namespace chip;
+using namespace chip::Access;
+using namespace chip::app::Clusters;
+
+namespace {
+// TODO: Maybe consider making this configurable?  See also
+// CHIPIMDispatch.mm.
+constexpr EndpointId kSupportedEndpoint = 0;
+
+// TODO: Make the policy more configurable by consumers.
+class AccessControlDelegate : public Access::AccessControl::Delegate {
+    CHIP_ERROR Check(
+        const SubjectDescriptor & subjectDescriptor, const RequestPath & requestPath, Privilege requestPrivilege) override
+    {
+        if (requestPath.endpoint != kSupportedEndpoint || requestPath.cluster != OtaSoftwareUpdateProvider::Id) {
+            // We only allow access to OTA software update provider.
+            return CHIP_ERROR_ACCESS_DENIED;
+        }
+
+        if (requestPrivilege != Privilege::kOperate) {
```

Comment:
If you also need `view` privilege, you'll have to explicitly check that also.

Reply:
(Check could be called for attributes or events, as well as commands.)

Reply:
We don't; the only things on this cluster are commands that require Operate.